pages/nft/iscn/_iscnId.vue

```diff
@@ -241,70 +241,82 @@ export default class NFTTestMintPage extends Vue {
-    await Promise.all([
-      this.getISCNInfo().catch(err => {
-        // eslint-disable-next-line no-console
-        console.error(err);
-        this.setError('ISCN_NOT_FOUND')
-      }),
-      // eslint-disable-next-line no-console
-      this.getMintInfo().catch(err => console.error(err)),
-    ]);
-    this.getOgImage()
+    try {
+      await Promise.all([
+        this.getISCNInfo(),
+        this.getMintInfo(),
+      ])
+      this.getOgImage()
+    } catch (error) {
+      this.setError(error)
+    }
   }
 
   async doAction() {
-    this.errorType = ''
-    this.balance = await getAccountBalance(this.address) as string
-    if(this.balance === '0'){
-      this.toggleSnackbar(ErrorType.INSUFFICIENT_BALANCE)
-      return
-    }
-    this.isLoading = true
-    /* eslint-disable no-fallthrough */
-    switch (this.state) {
-      case 'create':
-        if (!this.isUserISCNOwner) {
-          this.toggleSnackbar(ErrorType.USER_NOT_ISCN_OWNER)
-          break
-        }
+    try {
+      this.errorType = ''
+      this.balance = await getAccountBalance(this.address) as string
+      if (this.balance === '0'){
+        throw new Error(ErrorType.INSUFFICIENT_BALANCE)
+      }
+      this.isLoading = true
+      /* eslint-disable no-fallthrough */
+      switch (this.state) {
+        case 'create':
+          if (!this.isUserISCNOwner) {
+            throw new Error(ErrorType.USER_NOT_ISCN_OWNER)
+          }
 
-        try {
           if (this.ogImageBlob) {
-            const arweaveFeeInfo = await this.estimateArweaveFee()
-            if (!this.ogImageArweaveId) {
-              if (!this.ogImageArweaveFeeTxHash) { await this.sendArweaveFeeTx(arweaveFeeInfo) }
-              await this.submitToArweave()
+            try {
+              const arweaveFeeInfo = await this.estimateArweaveFee()
+              if (!this.ogImageArweaveId) {
+                if (!this.ogImageArweaveFeeTxHash) { await this.sendArweaveFeeTx(arweaveFeeInfo) }
+                await this.submitToArweave()
+              }
+            } catch (error) {
+              // eslint-disable-next-line no-console
+              console.error(error);
+              // skip uploading image if there is an error
             }
           }
           this.classId = await this.createNftClass()
           if (!this.classId) break
-        } catch (error) {
-          this.setError(error)
-        }
-
-      case 'mint':
-        this.sendRes = await this.mintNFT()
-        if (!this.sendRes) break
-        if (this.isWritingNFT) {
-          this.postInfo = await this.postMintInfo()
-          if (!this.postInfo) break
-          await this.getMintInfo()
-        }
-        this.isLoading = false
-        break
-      case 'done':
-        window.location.href = this.detailsPageURL
-      default:
+  
+        case 'mint':
+          await this.mintNFT()
+          if (!this.sendRes) break
+          if (this.isWritingNFT) {
+            await this.postMintInfo()
```

Comment:
I think the original way to write is clearer, you see `this.sendRes` assignment and its use in single screen
Any reason why you want to put it inside?

Reply:
Cuz I see other functions tend to assign props(`this.xxx`) inside the function., and inevitably check these props outside in the caller functions.
So I try to follow the style when refactoring.

Reply:
at lease the prop needs to be named to be related to the method I guess? `sendRes` is too general and cause confusion. `mintNFTResult` would be much more understandable without context